src/main/java/uk/gov/hmcts/darts/task/runner/impl/AbstractLockableAutomatedTask.java

```diff
@@ -36,7 +36,7 @@
 
 
 @Slf4j
-public abstract class AbstractLockableAutomatedTask implements AutomatedTask, AutoloadingAutomatedTask {
+public abstract class AbstractLockableAutomatedTask<T extends AbstractAutomatedTaskConfig> implements AutomatedTask, AutoloadingAutomatedTask {
```

Comment:
Aren't you overly complicating this by task instead of just having an interface that returns the config?

Reply:
@hemantasharma1129  what is your view?

Reply:
I have no issue on using Generics but does it not changes the scope of the ticket? Even though its not a big change of introducing Generics but does our test suite retests all the jobs?

Reply:
Our existing regression suite should cover the assurance from adding in generics, and the risk of this breaking anything is very low as any issues should be identified at compile time instead of runtime